test/integration/test_integration__io.py

```diff
@@ -285,6 +286,7 @@ def test_open_rasterio_mask_chunk_clip():
         )
         assert attrs == {
             "add_offset": 0.0,
+            "colorinterp": (rasterio.enums.ColorInterp.gray,),
```

Comment:
This will need to be converted to either a string representation or the integer representation. The reason for this is to enable calling `to_netcdf` without serialization issues.

Here is an example converting to the integer enumeration value:
```python
>>> import rasterio
>>> rasterio.enums.ColorInterp.gray
<ColorInterp.gray: 1>
>>> rasterio.enums.ColorInterp.gray.value
1
```

Reply:
An alternative would be to read/write the `photometric` creation option of the profile into/from the attributes. This is always in the string format and shouldn't have any serialization issues.

Reply:
Actually, the `colorinterp` enum can be serialized since it has an integer representation. It's the `colormap` dict which I'm having trouble with. The `colormap` dict looks something like this:

```python
{0: (248, 224, 168, 255),
 1: (240, 224, 152, 255),
 2: (104, 104, 96, 255),
 ...
 255: (248, 248, 248, 255)}
```
Do you think we should keep this `colormap` as an xarray attr, or should it be loaded in the data_var as 4 bands (i.e. RGBA)?

Reply:
Since it is a dict, you could use `json.dumps` & `json.loads` to convert to/from a string.

Reply:
Ok, the json serialization has been implemented in 54f9560a883fcfbbbc83a67b5abf7d431cca4384. Had to do some extra magic because the dict keys have to be `int`, but JSON only allows `str` keys. I did try an alternative method using yaml in 5b07d7fdc7b5a7d38fef89bca5795f26077ca710 as suggested by https://stackoverflow.com/questions/17099556/why-do-int-keys-of-a-python-dict-turn-into-strings-when-using-json-dumps#comment24737635_17099556, but maybe will stick with JSON.

Reply:
> Do you think we should keep this `colormap` as an xarray attr, or should it be loaded into the data_var as 4 bands (i.e. RGBA)?

Just to follow up on the data_var loading, I tried converting the [`earth_day_01d_p.tif`](https://oceania.generic-mapping-tools.org/server/earth/earth_day/earth_day_01d_p.tif) GeoTiff to NetCDF directly using `gdal_translate -of NetCDF -expand rgba earth_day_01d_p.tif earth_day_01d_p.nc`. 

This is the output of `gdalinfo earth_day_01d_p.nc`

```
Driver: netCDF/Network Common Data Format
Files: earth_day_01d_p.nc
Size is 512, 512
Metadata:
  NC_GLOBAL#Conventions=CF-1.5
  NC_GLOBAL#GDAL=GDAL 3.3.2, released 2021/09/01
  NC_GLOBAL#GDAL_AREA_OR_POINT=Area
  NC_GLOBAL#history=Sat Nov 06 23:41:47 2021: GDAL CreateCopy( earth_day_01d_p.nc, ... )
Subdatasets:
  SUBDATASET_1_NAME=NETCDF:"earth_day_01d_p.nc":Band1
  SUBDATASET_1_DESC=[180x360] Band1 (8-bit integer)
  SUBDATASET_2_NAME=NETCDF:"earth_day_01d_p.nc":Band2
  SUBDATASET_2_DESC=[180x360] Band2 (8-bit integer)
  SUBDATASET_3_NAME=NETCDF:"earth_day_01d_p.nc":Band3
  SUBDATASET_3_DESC=[180x360] Band3 (8-bit integer)
  SUBDATASET_4_NAME=NETCDF:"earth_day_01d_p.nc":Band4
  SUBDATASET_4_DESC=[180x360] Band4 (8-bit integer)
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0,  512.0)
Upper Right (  512.0,    0.0)
Lower Right (  512.0,  512.0)
Center      (  256.0,  256.0)
```
So the bands have been expanded into NetCDF subdatasets. I then loaded `earth_day_01d_p.nc` using rioxarray.open_rasterio, and it produces the following:

```
<xarray.Dataset>
Dimensions:  (y: 180, x: 360, band: 1)
Coordinates:
  * y        (y) float64 89.5 88.5 87.5 86.5 85.5 ... -86.5 -87.5 -88.5 -89.5
  * x        (x) float64 -179.5 -178.5 -177.5 -176.5 ... 176.5 177.5 178.5 179.5
  * band     (band) int64 1
    crs      int64 0
Data variables:
    Band1    (band, y, x) uint8 ...
    Band2    (band, y, x) uint8 ...
    Band3    (band, y, x) uint8 ...
    Band4    (band, y, x) uint8 ...
Attributes:
    Conventions:         CF-1.5
    GDAL:                GDAL 3.3.2, released 2021/09/01
    GDAL_AREA_OR_POINT:  Area
    history:             Sat Nov 06 23:41:47 2021: GDAL CreateCopy( earth_day...
```

Notice the Band1-Band4 data variables. So I'm thinking there should be a way to mimic this structure. Need to find a way to use the colormap dict currently in the attr, and get these Bands 1/2/3/4 information out as data variables.

Reply:
Mind doing `ncdump -h earth_day_01d_p.nc`  and pasting the output here?